cupy_setup_build.py

```diff
@@ -364,14 +365,7 @@ def module_extension_name(file):
 def module_extension_sources(file, use_cython, no_cuda):
     pyx, others = ensure_module_file(file)
     base = path.join(*pyx.split('.'))
-    if use_cython:
-        pyx = base + '.pyx'
-        if not os.path.exists(pyx):
-            use_cython = False
-            print(
-                'NOTICE: Skipping cythonize as {} does not exist.'.format(pyx))
-    if not use_cython:
-        pyx = base + '.cpp'
+    pyx = base + ('.pyx' if use_cython else '.cpp')
```

Comment:
I thought `.cpp` is removed?

Reply:
This is a hacky part (it's better to do refactoring). When doing `pip install cupy`:
1. setup.py: use `use_cython=False` to create a list of extension CPP sources generated by Cython (`*.cpp`)
2. setup.py: pass it to `setuptools.setup()` (at the bottom of the script)
3. setuptools: install Cython
5. cupy_setup_build.py: use `use_cython=True` to create a list of extension PYX sources to list up pyx files and  cythonize them (e.g., generate `core.cpp` from `core.pyx`).
6. setuptools: build the resulting CPP files, using the list generated in 1.

So `use_cython` is still needed here.